flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -527,12 +535,174 @@ public void testEagerSchedulingFailureReturnsSlot() throws Exception {
 	}
 
 	@Test
-	public void testPartitionRetainedWhenFinished() throws Exception {
+	public void testPartitionReleaseOnFinishWhileCanceling() throws Exception {
```

Comment:
can the code be deduplicated for these 2 tests?

Reply:
there are quite a few differences by now as to how the TE gateway is set up, what kind of tracker is used and how the job is constructed. Could we force deduplication? yes, but I don't think we'd make the test any more readable.